src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -1222,6 +1223,18 @@ protected virtual void AnalyzeTarget(TContext context, IEnumerable<Skimmer<TCont
 
         public static void AnalyzeTargetHelper(TContext context, IEnumerable<Skimmer<TContext>> skimmers, ISet<string> disabledSkimmers)
         {
+            // This might the public API invoked by other tools
+            Uri uri = context.CurrentTarget.Uri;
+            string originalStr = uri.OriginalString;
+            string filePath = uri.GetFilePath();
+            var URLPercentEncodingRegex = new Regex(@"%[0-9][0-9A-Fa-f]", RegexOptions.Compiled | RegexOptions.CultureInvariant);
```

Comment:
>URLPercentEncodingRegex [](http://example.com/codeflow?start=16&length=23)

forbidding URL percent-encoding isn't right.

this is a subtle topic, we need to talk about this more.

i think you will drop analysis of scan targets with your approach and we should be careful about that

Reply:
The new changes now allow scan targets with URL percent-encoding path. It won't be dropped anymore.